iothub/device/src/ClientPropertyCollection.cs

```diff
@@ -231,6 +231,40 @@ public bool Contains(string propertyName)
         /// <value>A <see cref="long"/> that is used to identify the version of the property collection.</value>
         public long Version { get; private set; }
 
+        /// <summary>
+        /// Gets the value of a component-level property.
+        /// </summary>
+        /// <remarks>
+        /// To get the value of a root-level property use <see cref="PayloadCollection.TryGetValue{T}(string, out T)"/>.
+        /// </remarks>
+        /// <typeparam name="T">The type to cast the object to.</typeparam>
+        /// <param name="componentName">The component which holds the required property.</param>
+        /// <param name="propertyName">The property to get.</param>
+        /// <param name="propertyValue">The value of the component-level property.</param>
+        /// <returns>true if the property collection contains a component level property with the specified key; otherwise, false.</returns>
+        public virtual bool TryGetValue<T>(string componentName, string propertyName, out T propertyValue)
+        {
+            if (Collection.ContainsKey(componentName))
+            {
+                // Extract the component-level properties into a JObject.
+                // The user specified serializer settings don't really affect this since this is an internal implementation detail.
```

Comment:
valid?

Reply:
Correct point, but based on our conversation earlier about this I like it being part of the Serializer instead of the collection. Getting feedback from the team will help guide this decision.